x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -111,6 +114,12 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
                     }
                     RemoteHostHeader.process(request, threadContext);
                     try {
+                        // Populate x-opaque-id if not already exists to chain all related actions together
+                        if (authentication != null && false == SystemUser.is(authentication.getUser())) {
+                            if (threadContext.getHeader(Task.X_OPAQUE_ID) == null) {
+                                threadContext.putHeader(Task.X_OPAQUE_ID, UUIDs.base64UUID());
+                            }
+                        }
```

Comment:
`X-Opaque-Id` automatically gets populate across related tasks and also cross nodes. This could be an alternative to chain all things together. It does not differentiate between parent and child. But it may not matter since the timestamp should tell most of the story?

Reply:
I like capturing this, if we have it we might as well use it - if nothing else, it provides an easy way to go find the trace for a request you just made if you're running this on a live system.

Reply:
Yes it is attached to every span in `ApmTracer#onTraceStarted`